ravenpy/models/importers.py

```diff
@@ -420,18 +379,191 @@ def extract(self) -> GridWeightsCommand:
 
                     area_intersect = inter.Area()
 
+                    area_all += area_intersect
+
                     if area_intersect > 0:
-                        hru_id = int(row[self._hru_id_field])
-                        cell_id = ilat * nlon + ilon
+                        hru_id = int(row[self._routing_id_field])
+                        cell_id = ilat * self._nlon + ilon
                         weight = area_intersect / area_basin
-                        grid_weights.append((hru_id, cell_id, weight))
+                        row_grid_weights.append((hru_id, cell_id, weight))
+
+            # mismatch between area of subbasin (routing product) and sum of all contributions of grid cells (model output)
+            error = (area_basin - area_all) / area_basin
+
+            if abs(error) > self._area_error_threshold and area_basin > 500000.0:
+                # record all basins with errors larger 5% (if basin is larger than 0.5 km2)
+                # error_dict[int(ibasin[key_colname])] = [error, area_basin]
+                grid_weights += row_grid_weights
+
+            else:
+                # adjust such that weights sum up to 1.0
+                for hru_id, cell_id, weight in row_grid_weights:
+                    corrected_weight = weight * 1.0 / (1.0 - error)
+                    grid_weights.append((hru_id, cell_id, corrected_weight))
+
+                # if error < 1.0:
+                #     area_all *= 1.0 / (1.0 - error)
+                # error = 0.0
 
         return GridWeightsCommand(
-            number_hrus=len(self._shapes),
-            number_grid_cells=nlon * nlat,
+            number_hrus=len(self._routing_data),
+            number_grid_cells=self._nlon * self._nlat,
             data=grid_weights,
         )
 
+    def _prepare_input_data(self):
+
+        if self._input_is_netcdf:
+
+            # Raven numbering is:
+            #
+            #      [      1      2      3   ...     1*nlon
+            #        nlon+1 nlon+2 nlon+3   ...     2*nlon
+            #           ...    ...    ...   ...     ...
+            #           ...    ...    ...   ...  nlat*nlon ]
+            #
+            # --> Making sure shape of lat/lon fields is like that
+            #
+
+            lon_var = self._input_data.variables[self._var_names[0]]
+            lon_dims = lon_var.dimensions
+            lon_var = lon_var[:]
+
+            lat_var = self._input_data.variables[self._var_names[1]]
+            lat_dims = lat_var.dimensions
+            lat_var = lat_var[:]
+
+            if len(lon_dims) == 2 and len(lat_dims) == 2:
+                if lon_dims == self._dim_names:
+                    lon_var = np.transpose(lon_var)
+
+                if lat_dims == self._dim_names:
+                    lat_var = np.transpose(lat_var)
+
+            elif len(lon_dims) == 1 and len(lat_dims) == 1:
+                # Coord vars are 1d, make them 2d
+                lon_var_size = lon_var.size
```

Comment:
needed?

Reply:
Yes because I reassign `lon_var` to itself on the next line, which would cause the line after the next (`lat_var = ..`) to be buggy, without this memorized value.

Reply:
Absolutely!